src/reviewer.py

```diff
@@ -0,0 +1,33 @@
+import os
+from github_utils import get_pull_request_diff, post_review_comment
+from llm_utils import analyze_code_changes
+
+class AICodeReviewer:
+    def __init__(self):
+        self.openai_api_key = os.getenv('GEMINI_API_KEY')
+        self.github_token = os.getenv('GIT_TOKEN')
+
+    def review_pull_request(self):
+        try:
+            # Get the PR diff
+            diff_content = get_pull_request_diff()
+            if not diff_content:
+                print("No diff content found")
+                return
+
+            # Analyze changes using LLM
+            review_comments = analyze_code_changes(diff_content)
+
+            # Post all comments in a single batch
+            if review_comments:
+                post_review_comment(review_comments, diff_content)
+            else:
+                print("No valid review comments generated")
+
+        except Exception as e:
+            print(f"Review failed: {str(e)}")
```

Comment:
The code currently prints "Review failed" and then re-raises the exception. This can lead to duplicate error messages. Consider either logging the error or re-raising, but not both, unless you want to handle it in the current scope *and* let the caller know about it.

```suggestion
Remove the `raise` statement if the intention is to simply log the error.
```